JITM: add an option to display an "activate" button in JITM.

jeherve opened the pull request:
#### Changes proposed in this Pull Request:

It allows one to activate a Jetpack module right from inside a JITM.

Companion diff: D19898-code

----------

**Without an activate_module parameter**:
<img width="1334" alt="screenshot 2018-10-25 at 16 37 32" src="https://user-images.githubusercontent.com/426388/47509197-e1734000-d875-11e8-9fa0-acca24a4b040.png">

**with**:

<img width="1285" alt="screenshot 2018-10-25 at 16 58 55" src="https://user-images.githubusercontent.com/426388/47509884-4bd8b000-d877-11e8-9b28-4c7844dd703f.png">


----------

@joanrho I would like to have your opinion on this because I am not entirely familiar with the different styles that should be applied to buttons depending on their site. Here is what we have so far. 
I do not believe we have anything in there that I could use to indicate that the change is *in progress*:
https://github.com/Automattic/jetpack/blob/master/scss/jetpack-admin-jitm.scss

#### Testing instructions:

1. Apply D19898-code to your WordPress.com sandbox.
2. Point your Jetpack site to your WordPress.com sandbox by adding `define( 'JETPACK__SANDBOX_DOMAIN', 'sandboxme.wordpress.com' );` to your Jetpack site's `wp-config.php` file (replace `sandboxme` by your own WordPress.com sandbox address). Also add `define( 'SCRIPT_DEBUG', true );` to that same file.
3. Point `jetpack.com` to your WordPress.com sandbox in your hosts file.
4. Go to `{my-site}/wp-admin/admin.php?page=jetpack_modules` and make sure that the Photon CDN module is off on your site.
5. Add the following to your site to make sure the CDN will not intervene while testing this PR: `add_filter( 'jetpack_force_disable_site_accelerator', '__return_true' );`
6. You're all set to start testing! Go to the main dashboard of your Jetpack site.

Make sure you can turn on the module from the JITM.

When you refresh the page, the JITM should be gone.

#### Proposed changelog entry for your changes:
* JITM: add an option to display an "activate" button in JITM.

jeherve committed:
JITM: add an option to display an "activate" button in JITM.

It allows one to activate a Jetpack module right from inside a JITM.

Companion diff: D19898-code

withinboredom commented on _inc/jetpack-jitm.js:
```diff
@@ -31,9 +31,22 @@ jQuery( document ).ready( function( $ ) {
+				html += '</div>';
+			}
 			if ( envelope.CTA.message ) {
+				var ctaClasses = 'jitm-button is-compact jptracks';
```
❤️

withinboredom reviewed (commented):
Some suggested changes, but otherwise looks good. Will test later today.

jeherve committed:
JITM: disable the button while the module is being activated.

jeherve committed:
JITM: only look for an activation button in our JITM.

@see https://github.com/Automattic/jetpack/pull/10400#discussion_r228253955

jeherve commented on _inc/jetpack-jitm.js:
```diff
@@ -80,6 +93,20 @@ jQuery( document ).ready( function( $ ) {
+				}
+			} ).done( function() {
+				$( '#jitm-banner__activate a' ).text( window.jitm_config.activated_module_text );
+				$( '#jitm-banner__activate a' ).attr( 'disabled', true );
```
Good call. Done in 5d36d5d7ba461df35331ecf615fd93b03a708740

jeherve commented on _inc/jetpack-jitm.js:
```diff
@@ -80,6 +93,20 @@ jQuery( document ).ready( function( $ ) {
 		// Add to Jetpack notices within the Jetpack settings app.
 		$template.prependTo( $( '#jp-admin-notices' ) );
+
+		$( '#jitm-banner__activate a' ).click( function() {
```
Makes sense. Done in c84f3f5f5545c1fbf5fe982a5d51b20068db4dee

brbrr reviewed (dismissed):
Tested well 👍 :shipit:

brbrr commented:
Not a big deal, but I noticed that after activating the module  - `Activated` button remains clickable which sends an activation request.

jeherve commented:
Hm. It should switch to `disabled`. Could you see the style changes?

brbrr commented:
> Hm. It should switch to `disabled`. Could you see the style changes?

Yeah, it looks like a disabled button, although - clickable

withinboredom commented:
I think you also have to add a `disabled` attribute.

jeherve committed:
JITM: do not allow clicking on Activation button if disabled.

@see https://github.com/Automattic/jetpack/pull/10400#issuecomment-433499425

The button had the disabled attribute already, but this was not stopping any onclick event from happening.

jeherve commented:
I fixed that `disabled` behaviour in 05b4d4dee96cc85db894044a1c24c056a1ba0517. It should work as expected now. 👍

dereksmart reviewed (approved):
Tests well! Haven't looked at the phab diff yet, but this PR looks good to go.
